app/models/guest_device.rb

```diff
@@ -10,4 +11,7 @@ class GuestDevice < ApplicationRecord
 
   has_one :network, :foreign_key => "device_id", :dependent => :destroy, :inverse_of => :guest_device
   has_many :miq_scsi_targets, :dependent => :destroy
+
+  has_many :firmwares, :dependent => :destroy
+  has_many :child_devices, -> { where(:parent_device_id => ids) }, :foreign_key => "parent_device_id", :class_name => "GuestDevice", :dependent => :destroy
```

Comment:
@blomquisg @juliancheal Could we just revert to my original changes since it does not have this problem and then create another PR to implement your suggestion (and look more into this issue)? This would help us because our upstream is broken since the provider changes are merged, but these changes are not. It's causing refresh to fail to discover hardware. Merging this PR would fix it. Including @rodneyhbrown7 on this discussion